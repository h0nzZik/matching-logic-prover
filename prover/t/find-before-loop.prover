claim \implies( \and( find-list( variable("H0") { ArrayIntInt }
                               , variable("X") { Int }
                               , variable("F0") { Set }
                               , .Patterns
                               )
                    , \equals( variable("X") { Int }
                               , variable("OLDX") { Int } )
                    , \equals( variable("TMP") { Int }, 0 )
                    , .Patterns
                    )
              , \and( find-list-seg( variable("H0") { ArrayIntInt }
                                   , variable("OLDX") { Int }
                                   , variable("X") { Int }
                                   , variable("F1") { Set }
                                   , .Patterns
                                   )
                    , disjoint(variable("F1") { Set }, variable("F2") { Set })
                    , \not(isMember(variable("DATA") { Int } , variable("F1") { Set }))
                    , .Patterns )
              )
strategy right-unfold-Nth(0,0) ; direct-proof

