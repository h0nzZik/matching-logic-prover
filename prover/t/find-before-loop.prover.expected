<prover>
  <k>
    .
  </k>
  <goals>
    <goal>
      <id>
        root
      </id>
      <active>
        true
      </active>
      <parent>
        .
      </parent>
      <claim>
        \implies ( \and ( find-list ( variable ( "H0" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "F0" ) { SetInt } , .Patterns ) , \equals ( variable ( "X" ) { Int } , variable ( "OLDX" ) { Int } ) , \equals ( variable ( "TMP" ) { Int } , 0 ) , .Patterns ) , \exists { variable ( "F1" ) { SetInt } , variable ( "F2" ) { SetInt } , variable ( "DATA" ) { Int } , .Patterns } \and ( find-list-seg ( variable ( "H0" ) { ArrayIntInt } , variable ( "OLDX" ) { Int } , variable ( "X" ) { Int } , variable ( "F1" ) { SetInt } , .Patterns ) , disjoint ( variable ( "F1" ) { SetInt } , variable ( "F2" ) { SetInt } , .Patterns ) , \not ( isMember ( variable ( "DATA" ) { Int } , variable ( "F1" ) { SetInt } , .Patterns ) ) , .Patterns ) )
      </claim>
      <strategy>
        success
      </strategy>
      <trace>
        .
      </trace>
    </goal>
  </goals>
</prover>
